src/metabase/api/collection.clj

```diff
@@ -146,22 +156,32 @@
     :pulse      Pulse
     :snippet    NativeQuerySnippet))
 
+(defn- pinned-state->pred [pinned-state]
+  (case pinned-state
+    :all (constantly true)
+    :is_pinned #(some? (% :collection_position))
```

Comment:
I'd recommend against using `%` as a function here... it works fine if the arg is a map but if it's `nil` you're doing `(nil :collection_position)` which will throw an NPE. `(:collection_position %)` on the other hand will happily accept `nil`.

It probably doesn't matter here but when there's a choice between a more robust way of doing the same thing and one with gotchas I'd go with the former